include/envoy/event/scaled_range_timer_manager.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include "envoy/common/pure.h"
+#include "envoy/event/range_timer.h"
+#include "envoy/event/timer.h"
+
+namespace Envoy {
+namespace Event {
+
+/**
+ * Describes a minimum timer value that is equal to a scale factor applied to the maximum.
+ */
+struct ScaleFactor {
+  explicit ScaleFactor(double scale_factor) : scale_factor_(scale_factor) {}
```

Comment:
This name seems confusing given the setScaleFactor() method below.  Consider alternate names: ScaledMin, AbsoluteMin (or ScaledMinimum/AbsoluteMinimum)